classes/migration/upgrade/PKPv3_3_0UpgradeMigration.inc.php

```diff
@@ -474,6 +474,18 @@ private function _migrateSubmissionFiles() {
 			->where('setting_name', 'fileRevision')
 			->update(['setting_name' => 'fileId']);
 
+		// Update file name of dependent files, see: pkp/pkp-lib#6801
+		$dependentFiles = Capsule::table('submission_files')
+			->select('file_id', 'original_file_name')
+			->where('file_stage' , '=', SUBMISSION_FILE_DEPENDENT)
+			->get();
+		foreach ($dependentFiles as $dependentFile) {
+			Capsule::table('submission_file_settings')
+				->where('file_id', '=', $dependentFile->file_id)
+				->where('setting_name', '=', 'name')
+				->update(['setting_value' => $dependentFile->original_file_name]);
+		}
```

Comment:
This might cause an issue with a large array as Alec mentioned. @asmecher is there a more performant way to do this?

Reply:
Something like https://laravel.com/docs/8.x/queries#chunking-results?

Reply:
I mean instead of getting all records in `$dependentFiles`, process by 100 items at a time.  I think, recalling the conversation, it should be enough.

Reply:
This wouldn't be a problem with array size, as `foreach(Capsule::table(...)->...->get())` will work through the result set via iteration rather than materializing to a PHP array. (This is where `get()` is better than `pluck()` -- the latter materializes the results into memory.)

It would be quicker if written as a multi-table `UPDATE`, but that would need to be written once each for PostgreSQL and MySQL and that's best avoided unless the optimization is necessary.

The large installations I'm aware of don't publish often with HTML, so I'd be happy to leave this as it is until we know it needs optimizing.

Reply:
Oh! Doing a bit of searching, it looks like `get()` may not actually iterate, but instead materializes into memory just like `pluck`. (See e.g.: https://stackoverflow.com/questions/28082279/how-do-i-use-laravels-chunk-to-avoid-running-out-of-memory) So yes, `chunk` is what I'd recommend using in large result lists -- it's easy enough to add here and I'll look at adding it elsewhere.